modules/core/shared/src/main/scala/coursier/core/DependencyManagement.scala

```diff
@@ -13,6 +13,10 @@ object DependencyManagement {
     `type`: Type,
     classifier: Classifier
   ) {
+
+    override lazy val hashCode: Int =
+      tuple.hashCode()
```

Comment:
how about `if (cachedHashCode ne null) cachedHashCode else cachedHashCode = tuple.hashCode`

Reply:
That might work too